.github/workflows/tests.yml

```diff
@@ -0,0 +1,76 @@
+name: Automated Tests
+on: [push]
+env:
+  NODE_OPTIONS: --max-old-space-size=5120
+jobs:
+  Contract:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v2
+      - name: Building on Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: '14.17.1'
+      - name: Update NPM
+        run: npm install -g npm@7.18.1
+      - uses: actions/cache@v2
+        with:
+          path: '**/node_modules'
+          key: ${{ runner.os }}-modules-${{ hashFiles('**/package-lock.json') }}
+      - name: Install dependencies
+        run: npm i --no-audit
+      - name: Compile contracts
+        run: npm run compile
+  Foundry_Tests:
+    runs-on: ubuntu-latest
+    env:
+      ARCHIVE_NODE_URL: ${{ secrets.ARCHIVE_NODE_URL_L2 }}
+      ETHERSCAN_API_KEY: ${{ secrets.ETHERSCAN_API_KEY }}
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v2
+      - name: Building on Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: '14.17.1'
+      - name: Update NPM
+        run: npm install -g npm@7.18.1
+      - uses: actions/cache@v2
+        with:
+          path: '**/node_modules'
+          key: ${{ runner.os }}-modules-${{ hashFiles('**/package-lock.json') }}
+      - name: Install dependencies
+        run: npm i --no-audit
+      - name: Compile contracts
+        run: npm run compile
+      - name: Install Foundry
+        uses: foundry-rs/foundry-toolchain@v1
+        with:
+          version: nightly
+      - name: Run tests
+        run: forge test --fork-url ${{ secrets.ARCHIVE_NODE_URL_L2 }} --fork-block-number 8626683 --etherscan-api-key ${{ secrets.ETHERSCAN_API_KEY }}
```

Comment:
Curious what the etherscan-api-key is for?

Reply:
Foundry allows for deeper stack tracing with the help of Etherscan's API